### DISCUSSION TITLE
How to fetch the WheelEvent or is there an alternative method ?

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#3364] [Q&A] [answered]
I'm very sorry that I have encountered some problems again.

I am writing a screenshot program, and the program can pin the image on the screen.
Now, I want to zoom the image by scrolling the wheel.

However, I found some relevant information: https://github.com/slint-ui/slint/issues/1280, https://github.com/slint-ui/slint/discussions/680, there maybe no public api to fetch the WheelEvent. Is there an alternative method?

I think https://github.com/slint-ui/slint/issues/1280 is a good idea, or add a wheel-event callback to Flickable?

**Comment by ogoffart** [comment, marked as answer]
We need to implement wheel event as described in #1280.

On what kind of control are you trying to intercept the wheel events?

**Reply by Horbin-Magician** [reply]
This is my old code when using qt and c++:

```c++
void ShotterWindow::wheelEvent(QWheelEvent *e)
{
    if(e->angleDelta().y() > 0) m_zoom = m_zoom + 0.1;
    else if (e->angleDelta().y() < 0 && m_zoom > 0.1) m_zoom = m_zoom - 0.1;
    setGeometry(zoomRect(m_geoRect, m_zoom).toRect());
    emit sgn_rect_change(this->geometry());
}
```

I don't know how to intercept the wheel events using slint.

**Reply by ogoffart** [reply]
One would need to implement #1280

**Reply by Horbin-Magician** [reply]
Yes, that's the easiest way. Thank you for your reply and thank you very much for the work of your team.
